receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -477,14 +477,19 @@ func (id *inFlightData) consumeDone(ctx context.Context, consumeErrPtr *error) {
 		id.span.SetStatus(otelcodes.Error, retErr.Error())
 	}
 
-	id.replyToCaller(retErr)
+	id.replyToCaller(ctx, retErr)
 	id.anyDone(ctx)
 }
 
-func (id *inFlightData) replyToCaller(callerErr error) {
-	id.pendingCh <- batchResp{
+func (id *inFlightData) replyToCaller(ctx context.Context, callerErr error) {
+	select {
+	case id.pendingCh <- batchResp{
 		id:  id.batchID,
 		err: callerErr,
+	}:
+		// OK: Responded.
+	case <-ctx.Done():
+		// OK: Never responded due to cancelation.
```

Comment:
Hmm how does this affect the read on the exporter side https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/otelarrowexporter/internal/arrow/stream.go#L201 - i.e. what error will be returned from the call to Recv on the exporter side?

Reply:
I'm going to try to answer this firmly, with a new test. I believe the client will have a canceled context any time the server does, so the client's context tree will eventually cause the client to cancel its work.

Reply:
@moh-osman3 Please see the new test which reproduces the case where if a client never calls Recv(), Shutdown can hang because it uses gRPC's graceful shutdown mechanism, and there is nothing to cancel the stream.

[e2ad461](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34179/commits/e2ad461dbe36c34384cfe65827d304f1ef1733eb)